### ISSUE TITLE
`guess_spec()` does not calculate `.required` correctly for tibbles with list of tibble columns

### ISSUE THREAD

**Issue by mgirlich** [mgirlich/tibblify#70]
``` r
library(tibble)
library(tibblify)

x <- list(
  list(x = list(list(x = 1))),
  list(x = list(list(x = 1, y = 1)))
)
# `y` is not required
guess_spec(x)
#> spec_df(
#>   x = tib_df(
#>     "x",
#>     x = tib_dbl("x"),
#>     y = tib_dbl("y", required = FALSE)
#>   )
#> )

x2 <- tibble(
  x = list(
    tibble(x = 1),
    tibble(x = 1, y = 1)
  )
)
# `y` is - incorrectly - required
guess_spec(x2)
#> spec_df(
#>   x = tib_df(
#>     "x",
#>     x = tib_dbl("x"),
#>     y = tib_dbl("y")
#>   )
#> )
```

<sup>Created on 2022-06-15 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by krlmlr** [comment]
Solved by #65?

**Comment by mgirlich** [comment]
Fixed by [670f112](https://github.com/mgirlich/tibblify/commit/670f112b030d2926df094e1587ca13f69baab5a3)
